src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -497,6 +500,19 @@ public void on(final ArrowLooseEvent event)
         checkEventCancelation(Action.SHOOT_ARROW, event.getPlayer(), event.getPlayer().getEntityWorld(), event, new BlockPos(event.getPlayer().getPositionVec()));
     }
 
+    @SubscribeEvent
+    public void on(final LivingHurtEvent event)
+    {
+        if (event.getEntity() instanceof ServerPlayerEntity
+              && event.getSource() instanceof EntityDamageSource
+              && event.getSource().getTrueSource() instanceof EntityCitizen
+              && ((EntityCitizen) event.getSource().getTrueSource()).getCitizenColonyHandler().getColonyId() == colony.getID()
+              && colony.getRaiderManager().isRaided())
+        {
```

Comment:
I think we want to check the players permissions with the colony here, probably the guard attack one. Also a better function name + javadoc above would be nice :)